app/controllers/rides_controller.rb

```diff
@@ -0,0 +1,66 @@
+# app/controllers/rides_controller.rb
+class RidesController < ApplicationController
+  # before_action :authenticate_user!
+
+  # GET /rides
+  def index
+    @rides = Ride.all
+    render json: @rides
+  end
+
+  # GET /rides/:id
+  def show
+    @ride = Ride.find(params[:id])
+    render json: @ride
+  end
+
+  # POST /rides
+  def create
+    @ride = Ride.new(ride_params)
+    if @ride.save
+      render json: @ride, status: :created
+    else
+      render json: @ride.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /rides/:id
+  def update
+    @ride = Ride.find(params[:id])
```

Comment:
This code is repeating, you can create new method and call that wherever required:
```ruby
def ride
  @ride ||= Ride.find(params[:id])
  render json: { error: "Ride not found" },  status: :not_found  if @ride.blank?
end
```

Reply:
Or you can raise exception